keyboards/gmmk/numpad/readme.md

```diff
@@ -0,0 +1,29 @@
+# GMMK NUMPAD
+
+A 17 Key macropad made ans old by Glorious LLC, equiped with a WB32 microcontroller, with support for a rotary encoder.
+
+* Keyboard Maintainer: [GloriousThrall](https://github.com/GloriousThrall)
+* Hardware Supported: GMMK Numpad
+* Hardware Availability: [GloriousPCGaming.com](https://www.gloriousgaming.com/products/gmmk-numpad)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make gmmk/numpad:default
+
+Flashing example for this keyboard:
+
+    make gmmk/numpad:default:flash
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
+## Bootloader
+
+Enter the bootloader in 3 ways:
+
+* **Bootmagic reset**: Hold down the Num key and plug in the keyboard
+* **Keycode in layout**: Press the Num+Del key which is mapped to `QK_BOOT`
+
+## Broken
+
+* Slider does not function
```

Comment:
There are a couple of keyboards with sliders in QMK (`keebwerk/nano_slider`, `ungodly/launch_pad`); they map the slider to a MIDI controller input, which then can be used by some software on the host side.  However, I can understand that you may not want to enable MIDI support in the default firmware due to compatibility concerns.

One option is adding a separate keymap which supports the slider by mapping it to MIDI.

Reply:
I've attempted this however I get an error regarding not enough endpoints.

Reply:
Ugh, apparently WB32F3G71 has only 3 bidirectional endpoints, like STM32F4x1.  So you may need the same workaround if you want to fit VIA + MIDI there:
```make
KEYBOARD_SHARED_EP = yes
```
This may bring some compatibility issues though, because the 6KRO keyboard protocol will now go over the same endpoint as NKRO, media keys and mouse emulation.  However, many STM32F4x1-based boards use that option.

(No idea whether WB32F3G71 has other specific compatibility issues though — I don't have any devices with that chip.)

Reply:
I will add it as a separate keymap for now, for some reason adding it to the default map will cause it to not build anymore.

Reply:
I'm getting, even when I add analog.c and analog.h
```
| 
 | /usr/lib/gcc/arm-none-eabi/12.2.0/../../../../arm-none-eabi/bin/ld: /tmp/ccgN9wGW.ltrans0.ltrans.o: in function `matrix_scan.isra.0':
 | /home/cameronv/projects/qmk_rustedaperture/./keyboards/gmmk/numpad/keymaps/default/keymap.c:52: undefined reference to `analogReadPin'
 | collect2: error: ld returned 1 exit status
 | 
make[1]: *** [builddefs/common_rules.mk:267: .build/gmmk_numpad_default.elf] Error 1
make: *** [Makefile:392: gmmk/numpad:default] Error 1
Make finished with errors
```

Reply:
For the analog support to work, you need to add `SRC += analog.c`, and also add `halconf.h` with `#define HAL_USE_ADC TRUE` and `mcuconf.h` with `#define WB32_ADC_USE_ADC1 TRUE` (you can find a sample in `keyboards/handwired/onekey/evb_wb32f3g71/`, although those files enable more peripherals).  Also you may need to run `qmk clean` after adding those files (the addition of extra files might not be properly picked up by the build system).

Reply:
Thank you, got the slider working now.